accio-tests/src/test/java/io/accio/testing/TestDuckDBResource.java

```diff
@@ -0,0 +1,165 @@
+            Manifest initial = Manifest.builder()
+                    .setCatalog("memory")
+                    .setSchema("tpch")
+                    .build();
+            Files.write(accioMDLFilePath, MANIFEST_JSON_CODEC.toJsonBytes(initial));
+
+            settingDir = Files.createTempDirectory("duckdb-setting");
+
+            ImmutableMap.Builder<String, String> properties = ImmutableMap.<String, String>builder()
+                    .put("accio.datasource.type", DUCKDB.name())
+                    .put("accio.directory", mdlDir.toAbsolutePath().toString())
+                    .put("duckdb.connector.setting-dir", settingDir.toAbsolutePath().toString());
+
+            return TestingAccioServer.builder()
+                    .setRequiredConfigs(properties.build())
+                    .build();
+        }
+        catch (IOException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    @AfterMethod
+    public void cleanTempDir()
+            throws IOException
+    {
+        Files.walk(settingDir)
+                .map(Path::toFile)
+                .forEach(File::delete);
+    }
+
+    @Test
+    public void testInitSQLWithFailedSQL()
+    {
+        assertWebApplicationException(() -> setDuckDBInitSQL(FAILED_SQL))
+                .hasErrorMessageMatches(".*Parser Error: syntax error at or near \"xxx\"");
+    }
+
+    @Test
+    public void testInitSQL()
+    {
+        assertThatCode(() -> setDuckDBInitSQL(INIT_SQL)).doesNotThrowAnyException();
+        assertThat(getDuckDBInitSQL()).isEqualTo(INIT_SQL);
+        assertThat(queryDuckDB(SHOW_TABLES_SQL))
+                .extracting(DuckDBQueryDto::getRows)
+                .isNotNull()
+                .asList().element(0)
+                .isEqualTo(ImmutableList.of("customer"));
+
+        assertThatCode(() -> appendToDuckDBInitSQL(APPEND_INIT_SQL)).doesNotThrowAnyException();
+        assertThat(getDuckDBInitSQL()).isEqualTo(INIT_SQL + "\n" + APPEND_INIT_SQL);
+        assertThat(queryDuckDB(SHOW_TABLES_SQL))
+                .extracting(DuckDBQueryDto::getRows)
+                .isNotNull()
+                .asList()
+                .isEqualTo(ImmutableList.of(ImmutableList.of("customer"), ImmutableList.of("orders")));
+
+        assertThatCode(() -> setDuckDBInitSQL(INIT_SQL)).doesNotThrowAnyException();
+        assertThat(getDuckDBInitSQL()).isEqualTo(INIT_SQL);
+        assertThat(queryDuckDB(SHOW_TABLES_SQL))
+                .extracting(DuckDBQueryDto::getRows)
+                .isNotNull()
+                .asList().element(0)
+                .isEqualTo(ImmutableList.of("customer"));
+    }
+
+    @Test
+    public void testSessionSQLWithFailedSQL()
+    {
+        assertWebApplicationException(() -> setDuckDBSessionSQL(FAILED_SQL))
+                .hasErrorMessageMatches(".*Parser Error: syntax error at or near \"xxx\"");
+    }
+
+    @Test
+    public void testSessionSQL()
```

Comment:
We should have a test to assert if the session SQL is applied to every exist connection in the pool.